zulipterminal/ui_tools/boxes.py

```diff
@@ -676,6 +673,7 @@ def transform_content(self) -> Tuple[None, Any]:
             padding_char = '░'
             self.indent_quoted_content(soup, padding_char)
 
+        self.link_index = 1
```

Comment:
For maintaining a link counter, I couldn't make a recursive approach work and eventually had to go with an attribute. I could use pointers on how to make it clearer.

Reply:
Could we use `state` for this?

Reply:
(Resolved in PM.)